HyperPlay-Gaming/hyperplay-desktop-client
# fix(overlay): set autoBind true to AuthState makeAutoObservable

<--- Put the description here --->

### Summary

This was happening due to the context of `this` being lost in this function handler:

```jsx
<QuestDetailsWrapper
   ...
    openSignInModal={authState.openSignInModal}
/>
```

See https://stackoverflow.com/a/69100459 for reference.


### How to test:

On Production:
- While being logged out click the sign in button from the overlay.
![image](https://github.com/user-attachments/assets/8442416c-8b8d-4031-a211-2fd701a83a08)
- Check that nothing happens.

In the terminal, you'll see this error:
```
(14:29:02) ERROR:   [Frontend]:         TypeError: Cannot set properties of undefined (setting 'signInModalOpen')
    at openSignInModal (http://localhost:5173/src/frontend/state/authState.ts:27:26)
    at executeAction (http://localhost:5173/node_modules/.vite/deps/chunk-4TF65JMT.js?v=526fa203:1117:15)
    at openSignInModal (http://localhost:5173/node_modules/.vite/deps/chunk-4TF65JMT.js?v=526fa203:1102:12)
    at HTMLUnknownElement.callCallback2 (http://localhost:5173/node_modules/.vite/deps/chunk-I3COAS7K.js?v=526fa203:3674:22)
```

With these changes, the login modal should pop up.

---

Use the following Checklist if you have changed something on the Backend or Frontend:
